### DISCUSSION TITLE
Use CadQuery for schematics?

### DISCUSSION THREAD

**Discussion by andlekbra** [CadQuery/cadquery#1687] [Q&A]
Hi

I am looking for a solution to programmatically generate schematics like simple p&ids, topology drawings, connection diagrams etc..
Are there any examples of CadQuery used for this purpose or is it not suitable for the task?

**Comment by jmwright** [comment]
I'm not aware of anyone using CadQuery for schematics, and while possible, is probably not going to be easy. I have experimented a little bit with [Skidl](https://devbisme.github.io/skidl/) for creating schematics using Python and [pcbflow](https://github.com/michaelgale/pcbflow) for routing, with limited success. I think my biggest problem was that I could not get the auto-layout to work correctly for schematics. While the nets were technically correct, the schematic got to be a mess for some schematics I was doing. I could not mimic how a human would lay out the schematic.

**Reply by andlekbra** [reply]
Thanks for the reply!
Noted. I will probably start experimenting with [drawpyo](https://github.com/MerrimanInd/drawpyo)
